### DISCUSSION TITLE
Compile pico-sdk locally

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#1811] [General]
When I understand this correctly, the pico-sdk shipped with the core is precompiled.
How can I change this to try out little changes and to have correct paths in the elf files for debugging purposes?
Is there an easy solutiuon?

**Comment by earlephilhower** [comment]
Assuming Linux or Linux-like environment where you've got `cmake` etc., look in `tools/libpico` for the docs and `make-libpico.sh` script.  (Note there is an SDK bug in one of the BOOT2.S files, fixed in `develop` but not released by the RPI guys, meaning you'll probably want to comment out the boot2 compilation stages).

Also, FWIW, there is a GDB command you can add to your init `.rc` file to regex replace paths in ELF files with your own locations.  No need to rebuild to get that working.
